Looks like copying from Arcade didn't help you:

``` text
MSBUILD : error MSB1001: Unknown switch.
Switch: --ResultsDirectory:/home/vsts/work/1/s/artifacts/TestResults/Release
```

--- a/.azure/pipelines/components-e2e-tests.yml
+++ b/.azure/pipelines/components-e2e-tests.yml
@@ -47,12 +47,12 @@ jobs:
     - script: .dotnet/dotnet test ./src/Components/test/E2ETest -c $(BuildConfiguration) --no-build --filter 'Quarantined!=true|Quarantined=false'
                  --logger:"trx%3BLogFileName=Microsoft.AspNetCore.Components.E2ETests.trx"
                  --logger:"html%3BLogFileName=Microsoft.AspNetCore.Components.E2ETests.html"
-                 "--ResultsDirectory:$(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)"
+                 --results-directory $(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)/Unquarantined
       displayName: Run E2E tests
     - script: .dotnet/dotnet test ./src/Components/test/E2ETest -c $(BuildConfiguration) --no-build --filter 'Quarantined=true' -p:RunQuarantinedTests=true
                  --logger:"trx%3BLogFileName=Microsoft.AspNetCore.Components.E2ETests.trx"
                  --logger:"html%3BLogFileName=Microsoft.AspNetCore.Components.E2ETests.html"
-                 "--ResultsDirectory:$(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)/Quarantined"
+                 --results-directory $(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)/Quarantined
       displayName: Run Quarantined E2E tests
       continueOnError: true
     - task: PublishTestResults@2
@@ -60,7 +60,7 @@ jobs:
       inputs:
         testResultsFormat: 'VSTest'
         testResultsFiles: '*.trx'
-        searchFolder: '$(Build.SourcesDirectory)/src/Components/test/E2ETest/TestResults'
+        searchFolder: '$(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)/Unquarantined'
         testRunTitle: ComponentsE2E-$(AgentOsName)-$(BuildConfiguration)-xunit
       condition: always()
     - task: PublishTestResults@2

--- a/.azure/pipelines/components-e2e-tests.yml
+++ b/.azure/pipelines/components-e2e-tests.yml
@@ -75,5 +75,5 @@ jobs:
 
     artifacts:
     - name: Components_E2E_Test_Logs
-      path: ./src/Components/test/E2ETest/TestResults
+      path: '$(Build.SourcesDirectory)/artifacts/TestResults/$(BuildConfiguration)'
       publishOnError: true

